test/unit/core/Object3D.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * @author simonThiele / https://github.com/simonThiele
+ */
+
+module( "Object3D" );
+
+test( "getWorldScale", function() {
+  var obj = new THREE.Object3D();
+
+  checkIfDecomposeAndGetWorldScaleAreEqual(obj);
+
+  obj.scale.set(1, 1, 1);
+  checkIfDecomposeAndGetWorldScaleAreEqual(obj);
+
+  obj.scale.set(-1, -12.4324, 32456323);
+  checkIfDecomposeAndGetWorldScaleAreEqual(obj);
+});
+
+test( "getWorldScale with hierarchy", function() {
+  var obj = new THREE.Object3D();
+  var parent1 = new THREE.Object3D();
+  var parent2 = new THREE.Object3D();
+
+  parent2.add(parent1);
+  parent1.add(obj);
+
+  checkIfDecomposeAndGetWorldScaleAreEqual(obj);
+
+  obj.scale.set(1, 1, 1);
+  parent2.scale.set(-1, -12.4324, 32456323);
+  parent1.scale.set(10, 20, -234);
+
+  checkIfDecomposeAndGetWorldScaleAreEqual(obj);
+
+  obj.scale.set(1, 0.00123, -1);
+  parent2.scale.set(-1, -12.4324, 32456323);
+  parent1.scale.set(10, 20, -234);
+
+  checkIfDecomposeAndGetWorldScaleAreEqual(obj);
+
+  obj.position.set(1, 0.00123, -1);
+  parent2.position.set(-1, -12.4324, 32456323);
+  parent1.position.set(10, 20, -234);
+
+  checkIfDecomposeAndGetWorldScaleAreEqual(obj);
+});
+
+function checkIfDecomposeAndGetWorldScaleAreEqual(obj) {
+  var result = new THREE.Vector3();
+
+  // this is the reference way for comparing. We decompose the hole matrix and
+  // extract the scale part
+  var position = new THREE.Vector3();
+  var quaternion = new THREE.Quaternion();
+
+  obj.updateMatrixWorld( true );
```

Comment:
You need to call `updateMatrixWorld()` on the parent of the object's parent.

I suggest that you test this by printing out values and checking them manually so you can verity that you are testing what you think you are testing.